src/eei.cpp

```diff
@@ -119,12 +118,108 @@ Literal EthereumInterface::callImport(Import *import, LiteralList& arguments) {
       memory.set<uint8_t>(j, src[i]);
     }
   }
+  
+  /*
+   * Memory Operations
+   */
 
-  void EthereumInterface::copyAddressToMemory(struct evm_address const& address, uint32_t dstoffset)
+  void EthereumInterface::loadMemory(uint32_t srcOffset, uint8_t *dst, size_t length)
   {
-    heraAssert(memory.size() < (dstoffset + 20), "Out of bounds (destination) memory copy.");
-    for (uint32_t i = 0, j = dstoffset; j < (dstoffset + 20); i++, j++) {
-      memory.set<uint8_t>(j, address.bytes[i]);
-    }
+      assert(dst != nullptr);
+      assert(length);
+
+      for (uint32_t i = 0; i < length; ++i) {
+          *(dst + i) = memory.get<uint8_t>(srcOffset + i);
+      }
+  }
+
+  void EthereumInterface::storeMemory(uint32_t dstOffset, uint8_t *src, size_t length)
+  {
+      assert(src != nullptr);
+      assert(length);
+
+      for (uint32_t i = 0; i < length; ++i) {
+          memory.set<uint8_t>(dstOffset + i, *(src + i));
+      }
+  }
+
+  /*
+   * Memory Op Wrapper Functions
+   */
+
+  void EthereumInterface::loadUint256(uint32_t srcOffset, struct evm_uint256be *dst)
```

Comment:
These should use constant references `const&` instead of pointers.

Reply:
This would break the load operations, since the function writes to the object referred to by the pointer.